src/main/java/jenkins/plugins/git/GitSCMMatrixUtil.java

```diff
@@ -0,0 +1,36 @@
+package jenkins.plugins.git;
+
+import hudson.matrix.MatrixBuild;
+import hudson.matrix.MatrixRun;
+import hudson.model.Run;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.Revision;
+import hudson.plugins.git.util.Build;
+import hudson.plugins.git.util.BuildData;
+
+import java.util.Collection;
+import java.util.Collections;
+
+public class GitSCMMatrixUtil {
```

Comment:
This entire thing could probably remain inline, but I like it better this way.

Of course the catching of `Throwable`s could (should? Classloading…) remain in `GitSCM`.